src/hooks/useAlerts.ts

```diff
@@ -49,28 +49,31 @@ export const defaultRules = {
   ],
 };
 
-const fetchSMRules = async (alertRulerUrl: string) => {
-  try {
-    return await getBackendSrv()
-      .fetch<any>({
-        method: 'GET',
-        url: `${alertRulerUrl}/rules/${SM_ALERTING_NAMESPACE}/default`,
-        headers: {
-          'Content-Type': 'application/yaml',
-        },
-      })
-      .toPromise()
-      .then((response) => {
-        const alertGroup = parse(response.data);
-        return alertGroup.rules;
-      });
-  } catch (e) {
-    if (e.status === 404) {
-      return [];
-    }
-    throw new Error(`Could not fetch alerting rules for Synthetic Monitoring`);
-  }
-};
+interface RuleResponse {
+  rules: AlertRule[];
+  error?: string;
+}
+
+const fetchSMRules = (alertRulerUrl: string): Promise<RuleResponse> =>
+  getBackendSrv()
+    .fetch<any>({
+      method: 'GET',
+      url: `${alertRulerUrl}/rules/${SM_ALERTING_NAMESPACE}/default`,
+      headers: {
+        'Content-Type': 'application/yaml',
+      },
+    })
+    .toPromise()
+    .then((response) => {
+      const alertGroup = parse(response.data);
+      return { rules: alertGroup.rules };
+    })
+    .catch((e) => {
+      if (e.status === 404) {
+        return { rules: [] };
```

Comment:
Not sure if I understand correctly, but based on this, the `alertRules` will always have an empty array, meaning in the `Alerting.tsx` component `{!alertRules && <Spinner />}` will spin even if an error is thrown.  

Probably a simple fix would be ` {!!alertRules.length && <Spinner />}`?

Reply:
Good catch!